apps/designer/app/designer/features/sidebar-left/panels/pages/settings.tsx

```diff
@@ -84,45 +168,186 @@ export const NewPageSettings = ({
-            </InputErrorsTooltip>
-          </Group>
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+            onSubmit();
+          }}
+        >
+          <FormFields {...formFieldsProps} />
           <Group css={{ alignItems: "end" }}>
             <Button type="submit" variant="green" disabled={isSubmitting}>
               {isSubmitting ? "Creating..." : "Create"}
             </Button>
           </Group>
-        </fetcher.Form>
+        </form>
+      </Box>
+    </>
+  );
+};
+
+const toFormData = (page: Partial<EditablePage> & { id: string }): FormData => {
+  const formData = new FormData();
+  for (const [key, value] of Object.entries(page)) {
+    // @todo: handle "meta"
+    if (typeof value === "string") {
+      formData.append(key, value);
+    }
+  }
+  return formData;
+};
+
+export const PageSettings = ({
+  onClose,
+  onDeleted,
+  pageId,
+  projectId,
+}: {
+  onClose?: () => void;
+  onDeleted?: () => void;
+  pageId: string;
+  projectId: string;
+}) => {
+  const submitPersistently = usePersistentFetcher();
+
+  const fetcher = useFetcher<EditPageData>();
+
+  const [pages] = usePages();
+  const page = pages && projectUtils.pages.findById(pages, pageId);
+
+  const [unsavedValues, setUnsavedValues] = useState<Partial<EditablePage>>({});
+  const [submittedValues, setSubmittedValues] = useState<Partial<EditablePage>>(
+    {}
+  );
+
+  const { fieldErrors, resetFieldError } = useFetcherErrors({
+    fetcher,
+    fieldNames,
+  });
+
+  const handleChange = useCallback(
+    <Name extends FieldName>(event: {
+      field: Name;
+      value: EditablePage[Name];
+    }) => {
+      resetFieldError(event.field);
+      setUnsavedValues((values) => ({ ...values, [event.field]: event.value }));
+    },
+    [resetFieldError]
+  );
+
+  useDebounce(
+    () => {
+      if (Object.keys(unsavedValues).length === 0) {
+        return;
+      }
+
+      // We're re-submitting the submittedValues because previous submit is going to be cancelled
+      // (normally, submittedValues are empty at this point)
+      const valuesToSubmit = { ...submittedValues, ...unsavedValues };
+
+      fetcher.submit(toFormData({ id: pageId, ...valuesToSubmit }), {
+        method: "post",
+        action: `/rest/pages/${projectId}`,
+      });
+
+      setSubmittedValues(valuesToSubmit);
+      setUnsavedValues({});
+    },
+    1000,
+    [unsavedValues]
+  );
+
+  useUnmount(() => {
+    if (Object.keys(unsavedValues).length === 0) {
+      return;
+    }
+    // We use submitPersistently instead of fetcher.submit
+    // because we don't want the request to be canceled when the component unmounts
+    submitPersistently<EditPageData>(
+      toFormData({ id: pageId, ...submittedValues, ...unsavedValues }),
+      { method: "post", action: `/rest/pages/${projectId}` },
+      (data) => {
+        if ("errors" in data) {
+          toastNonFieldErrors(normalizeErrors(data.errors), []);
```

Comment:
I remember I was confused about that name "toastNonFieldErrors" and had a suggestion

Reply:
Yeah. I answered there that `toastUnknownFieldErrors` wouldn't be ideal either, but thinking again it's not that important, will rename.